src/sandstorm/run-bundle.c++

```diff
@@ -1839,6 +1840,23 @@ private:
 
   pid_t startNode(const Config& config) {
     Subprocess process([&]() -> int {
+      int sockFd;
+      KJ_SYSCALL(sockFd = socket(PF_INET, SOCK_STREAM, IPPROTO_TCP));
+      sockaddr_in sa;
+      memset(&sa, 0, sizeof sa);
+      sa.sin_family = AF_INET;
+      sa.sin_port = htons(config.port);
+      inet_pton(AF_INET, config.bindIp.cStr(), &(sa.sin_addr));
+
+      KJ_SYSCALL(bind(sockFd, (struct sockaddr *)&sa, sizeof sa));
+      KJ_SYSCALL(listen(sockFd, 511)); // 511 is what node uses as its default backlog
+      KJ_SYSCALL(fcntl(sockFd, F_SETFL, fcntl(sockFd, F_GETFL, 0) | O_NONBLOCK));
```

Comment:
You can pass SOCK_NONBLOCK as a parameter to `socket()` above rather than use `fcntl()` here.